
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
        <link rel="prev" href="../labtat/">
      
      
        <link rel="next" href="../quantitative/">
      
      
      <link rel="icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.0, mkdocs-material-9.5.30">
    
    
      
        <title>Sequencing Analysis - NICD Wastewater Genomics Syndicate</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.3cba04c6.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="light-green" data-md-color-accent="purple">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#wastewater-genomics-sequencing-analysis-for-sars-cov-2" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="NICD Wastewater Genomics Syndicate" class="md-header__button md-logo" aria-label="NICD Wastewater Genomics Syndicate" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            NICD Wastewater Genomics Syndicate
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Sequencing Analysis
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="default" data-md-color-primary="light-green" data-md-color-accent="purple"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M17 6H7c-3.31 0-6 2.69-6 6s2.69 6 6 6h10c3.31 0 6-2.69 6-6s-2.69-6-6-6zm0 10H7c-2.21 0-4-1.79-4-4s1.79-4 4-4h10c2.21 0 4 1.79 4 4s-1.79 4-4 4zM7 9c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="slate" data-md-color-primary="light-green" data-md-color-accent="lime"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M17 7H7a5 5 0 0 0-5 5 5 5 0 0 0 5 5h10a5 5 0 0 0 5-5 5 5 0 0 0-5-5m0 8a3 3 0 0 1-3-3 3 3 0 0 1 3-3 3 3 0 0 1 3 3 3 3 0 0 1-3 3Z"/></svg>
      </label>
    
  
</form>
      
    
    
      <script>var media,input,key,value,palette=__md_get("__palette");if(palette&&palette.color){"(prefers-color-scheme)"===palette.color.media&&(media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']"),palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent"));for([key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
            
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  
  
    <li class="md-tabs__item">
      <a href=".." class="md-tabs__link">
        
  
    
  
  Home

      </a>
    </li>
  

      
        
  
  
  
    <li class="md-tabs__item">
      <a href="../sample_maps/" class="md-tabs__link">
        
  
    
  
  Sample Maps

      </a>
    </li>
  

      
        
  
  
  
    <li class="md-tabs__item">
      <a href="../labtat/" class="md-tabs__link">
        
  
    
  
  Turnaround Times

      </a>
    </li>
  

      
        
  
  
    
  
  
    <li class="md-tabs__item md-tabs__item--active">
      <a href="./" class="md-tabs__link">
        
  
    
  
  Sequencing Analysis

      </a>
    </li>
  

      
        
  
  
  
    <li class="md-tabs__item">
      <a href="../quantitative/" class="md-tabs__link">
        
  
    
  
  SARS-CoV-2 Quantitative Levels

      </a>
    </li>
  

      
        
  
  
  
    <li class="md-tabs__item">
      <a href="../influenza/" class="md-tabs__link">
        
  
    
  
  Influenza Analysis

      </a>
    </li>
  

      
    </ul>
  </div>
</nav>
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  


  

<nav class="md-nav md-nav--primary md-nav--lifted md-nav--integrated" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="NICD Wastewater Genomics Syndicate" class="md-nav__button md-logo" aria-label="NICD Wastewater Genomics Syndicate" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    NICD Wastewater Genomics Syndicate
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Home
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../sample_maps/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Sample Maps
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../labtat/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Turnaround Times
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  <span class="md-ellipsis">
    Sequencing Analysis
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  <span class="md-ellipsis">
    Sequencing Analysis
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#purpose" class="md-nav__link">
    <span class="md-ellipsis">
      Purpose
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#definitions" class="md-nav__link">
    <span class="md-ellipsis">
      Definitions
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#data-preprocessing" class="md-nav__link">
    <span class="md-ellipsis">
      Data Preprocessing
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Data Preprocessing">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#alignment" class="md-nav__link">
    <span class="md-ellipsis">
      Alignment
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#converting-sam-files-to-bam-file-and-sorting-the-bam-files" class="md-nav__link">
    <span class="md-ellipsis">
      Converting SAM files to BAM file, and sorting the BAM files**:
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#read-trimming-variant-calling-and-analysis" class="md-nav__link">
    <span class="md-ellipsis">
      Read Trimming, Variant Calling, and Analysis:
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#freyja-analysis" class="md-nav__link">
    <span class="md-ellipsis">
      Freyja Analysis:
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../quantitative/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    SARS-CoV-2 Quantitative Levels
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../influenza/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Influenza Analysis
  </span>
  

      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


<h1 id="wastewater-genomics-sequencing-analysis-for-sars-cov-2">Wastewater Genomics Sequencing Analysis For SARS-CoV-2</h1>
<h2 id="purpose">Purpose</h2>
<p>This documentation page outlines the procedure for processing and analyzing wastewater sequencing data as part of the wastewater genomics surveillance program. The goal is to ensure consistent, accurate, and reliable data for public health monitoring and decision-making.</p>
<h2 id="definitions">Definitions</h2>
<ul>
<li><strong>BWA</strong>: BWA is a software package used to map sequences with low divergence to a sizable reference genome, like the Wuhan-hu SARS-CoV-2 genome. Three algorithms make up this system: BWA-backtrack, BWA-SW, and BWA-MEM.</li>
<li><strong>Exatype</strong>: Exatype is a software platform developed by Hyrax Biosciences designed to analyze genetic sequences of viruses and other pathogens. Exatype could help in determining how these viruses are mutating and potentially inform the development of treatment strategies and vaccines by identifying mutations that might confer resistance to existing therapies.</li>
<li><strong>Freyja</strong>: Freyja is a tool to recover relative lineage abundances from mixed SARS-CoV-2 samples from a sequencing dataset (BAM aligned to the Hu-1 reference). The technique solves the limited (unit sum, non-negative) de-mixing problem by using lineage-determining mutational “barcodes” produced from the UShER global phylogenetic tree as a basis set.</li>
<li><strong>Mutational profile</strong>: Refers to the characterization of mutations present within a genome, gene, or set of genes in an organism. This profile includes the types, locations, and frequencies of genetic alterations, providing insights into the molecular underpinnings of various conditions, diseases, or evolutionary processes.</li>
<li><strong>NGS</strong>: Next-generation Sequencing, a method used for precise and comprehensive analysis of the genetic material found in wastewater samples.</li>
<li><strong>Samtools</strong>: Samtools is a set of tools used to interact with high-throughput sequencing data.</li>
<li><strong>iVar</strong>: iVar is a software package that consists of functions broadly useful for viral amplicon-based sequencing.</li>
<li><strong>Wastewater sample</strong>: A sample collected from various wastewater catchment sites, intended for genomic analysis.</li>
</ul>
<p>All commands/scripts must be executed on a Linux server/environment with Conda activated.</p>
<h2 id="data-preprocessing">Data Preprocessing</h2>
<p>Raw sequence data preprocessing steps including quality control, trimming of adapters, and removal of low-quality sequences are performed by the NICD Sequencing Core Facility and are therefore not included in this documentation page.</p>
<h3 id="alignment">Alignment</h3>
<ul>
<li><strong>Aligning reads to reference SARS-CoV-2 genome</strong>:</li>
</ul>
<p>This script performs alignment of raw sequencing reads to the reference SARS-CoV-2 genome using the BWA (Burrows-Wheeler Aligner) tool.
Before running, the input and output directories of all files are first defined. </p>
<p>This line defines the path to the reference genome sequence file.
The reference genome is the known sequence of the SARS-CoV-2 genome to which the raw sequencing reads will be aligned.</p>
<pre><code> # Directory where your indexed reference sequence is
REF=”/path/to/reference/sequence.fa
</code></pre>
<p>This line defines the directory where the raw sequencing files (in fastq.gz format) are located.
These are the files containing the raw reads obtained from sequencing the wastewater samples.</p>
<pre><code>RAW_DIR="/path/to/raw_files"
</code></pre>
<p>This line defines the output directory where the aligned SAM files will be saved. 
SAM files (Sequence Alignment/Map) store the alignment information of reads to a reference genome.</p>
<pre><code>OUT_DIR="./mapped_epi6_8"
</code></pre>
<p>This command creates the output directory defined in OUT_DIR if it does not already exist. 
The -p flag ensures that the command does not throw an error if the directory already exists.</p>
<pre><code>mkdir -p "$OUT_DIR"
</code></pre>
<p>A for loop is created to iterate over each R1 fastq.gz file in the raw files directory (RAW_DIR). R1 fastq.gz files contain the forward reads 
obtained from sequencing.The filename of the R1 file is then modified to generate the corresponding R2 filename. 
In Illumina paired-end sequencing, R1 and R2 files contain paired-end reads, where R1 contains the forward reads and R2 contains the reverse reads. 
This assumes a consistent naming convention where R1 files end with "_R1_001.fastq.gz" and R2 files end with "_R2_001.fastq.gz".</p>
<pre><code>for R1 in "$RAW_DIR"/*_R1_001.fastq.gz; do
# Assuming the naming convention is consistent, modify the suffix to find the R2 files
    R2="${R1/_R1_/_R2_}"
# Extract the sample name based on the R1 file
    SAMPLE_NAME=$(basename "$R1" "_R1_001.fastq.gz")
# Construct the output file name
    SAM_FILE="${OUT_DIR}/${SAMPLE_NAME}.aligned.sam"
</code></pre>
<p>This line prints a message indicating which sample is being mapped.</p>
<pre><code>echo "Mapping $SAMPLE_NAME..."
</code></pre>
<p>his line executes the BWA MEM algorithm to align the paired-end reads ($R1 and $R2) to the reference genome ($REF). 
The alignment results are then redirected (&gt;) to the output SAM file ($SAM_FILE).</p>
<pre><code>bwa mem "$REF" "$R1" "$R2" &gt; "$SAM_FILE"
</code></pre>
<p>This marks the end of the loop.</p>
<pre><code>done
</code></pre>
<p>This line prints a message indicating that the mapping process for all samples is complete.</p>
<pre><code>echo "Mapping complete."
</code></pre>
<h3 id="converting-sam-files-to-bam-file-and-sorting-the-bam-files">Converting SAM files to BAM file, and sorting the BAM files**:</h3>
<pre><code> ```bash
 # Script available in the documentation
 ```
</code></pre>
<h3 id="read-trimming-variant-calling-and-analysis">Read Trimming, Variant Calling, and Analysis:</h3>
<ul>
<li><strong>Trimming the reads and sorting</strong>:</li>
</ul>
<p>This script will loop through all indexed BAM files in the indexed directory and then extract the base filename without the .sorted.bam extension. Ultimately it will define the output trimmed BAM file path and output trimmed and sorted BAM file path</p>
<pre><code> for INDEXED_FILE in "$INDEXED_DIR"/*.sorted.bam; do
 BASENAME=$(basename "$INDEXED_FILE" .sorted.bam)
 TRIMMED_FILE="$TRIMMED_DIR/${BASENAME}.trimmed.bam"
</code></pre>
<p>This line will now trim the reads for each sample</p>
<pre><code> echo "Trimming reads for $INDEXED_FILE"
 ivar trim -b nCoV-2019_v1.bed -i "$INDEXED_FILE" -p "$TRIMMED_FILE" -q 15 -m 100 -s 4 -      e -x 3
</code></pre>
<p>This line will sort the trimmed reads</p>
<pre><code> echo "Sorting trimmed reads for $TRIMMED_FILE"
 samtools sort "$TRIMMED_FILE" -o "$TRIMMED_SORTED_FILE"
 done
 echo "Read trimming and sorting complete."
</code></pre>
<h3 id="freyja-analysis">Freyja Analysis:</h3>
<ul>
<li><strong>Running Freyja variants</strong>:</li>
<li>
<p>This script will specify the directory where the variants and depths files are located.</p>
<p>VARIANTS_DIR="/RAW/Depths_TSV_all_copy_B/TSV_all/TSV"
  DEPTHS_DIR="/RAW/Depths_TSV_all_copy_B/Depths_all/Depths"</p>
</li>
</ul>
<p>This will create the output directory (if not yet created) where the results will be stored</p>
<pre><code>  OUTPUT_DIR="/RAW/Depths_TSV_all_copy_B/output"
  mkdir -p "${OUTPUT_DIR}"
</code></pre>
<p>Loop into the variant files in the variants directory and then extract the base name for matching with the depth file. It will then construct the path to the matching depth file and subsequently specify the output file name, ending with .demix.tsv.</p>
<pre><code>  for variants_file in ${VARIANTS_DIR}/*.tsv; do
  base_name=$(basename "${variants_file}" .tsv)
  depth_file="${DEPTHS_DIR}/${base_name}.depths"
  output_file="${OUTPUT_DIR}/${base_name}.demix.tsv"
</code></pre>
<p>This script will check if the depth file exists(not compulsory) then run the Freyja variants step.</p>
<pre><code>  if [ ! -f "${depth_file}" ]; then
 echo "Depth file for ${base_name} does not exist. Skipping..."
 continue
 fi
 #Run Freyja demix
 echo "Processing: ${base_name}"
 freyja demix "${variants_file}" "${depth_file}" --output "${output_file}"
 done
 echo "All samples processed."
</code></pre>
<ul>
<li><strong>Freyja Demix</strong>:</li>
</ul>
<p>This script will iterate over each TSV file created on the previous step then it runs the Freyja demix step using the tsv file and depths file.</p>
<pre><code>   For tsv_file in /RAW/Depths_TSV_all_copy_B/TSV_all/TSV/*.tsv; do
   sample_name=$(basename “tsv_file” .tsv)
   depths_file= “/RAW/Depths_TSV_all_copy_B/Depths/${sample_name}.depths”
   output_file=”RAW/Freyja_demix/${sample_name}.demix.tsv
   echo “Running Freyja for $tsv_file”
   Freyja demix “$tsv_file” “$depths_file” –output “output_file”
   done.
</code></pre>
<ul>
<li>
<p><strong>Aggregating the Freyja Output Files</strong>:</p>
<p>freyja aggregate /RAW/Freyja_demix/ --ext .tsv –output                           /RAW/Freyja_demix/aggregate_file/agg_epi2_8.tsv</p>
</li>
</ul>
<h1 id="mutational-and-snps-analysis">Mutational and SNPs Analysis:</h1>
<ul>
<li><strong>Exatype Parameters</strong>:
     <code>Data Structure: Paired-end
     Sequencing Platform: Illumina
     Assay: ARTIC V4</code></li>
<li><strong>Generate Heatmap and Mutational Profile</strong>:
     <code>R
     # Script available in the documentation</code></li>
</ul>
<h2 id="references">References</h2>
<ul>
<li>Castellano, Sara, et al. “iVar, an Interpretation-Oriented Tool to Manage the Update and Revision of Variant Annotation and Classification.” <em>Genes</em> 12, no. 3 (March 8, 2021): 384. <a href="https://doi.org/10.3390/genes12030384">DOI: 10.3390/genes12030384</a></li>
<li>Dahui, Qin. “Next-Generation Sequencing and Its Clinical Application.” <em>Cancer Biology &amp; Medicine</em> 16, no. 1 (February 1, 2019): 4–10. <a href="https://doi.org/10.20892/j.issn.2095-3941.2018.0055">DOI: 10.20892/j.issn.2095-3941.2018.0055</a></li>
<li>Grubaugh, N.D., Gangavarapu, K., Quick, J. et al. An amplicon-based sequencing framework for accurately measuring intrahost virus diversity using PrimalSeq and iVar. Genome Biol 20, 8 (2019). https://doi.org/10.1186/s13059-018-1618-7</li>
<li>Karthikeyan, Smruthi, et al. “Wastewater Sequencing Reveals Early Cryptic SARS-CoV-2 Variant Transmission.” <em>Nature</em> 609, no. 7925 (September 1, 2022): 101–8. <a href="https://doi.org/10.1038/s41586-022-05049-6">DOI: 10.1038/s41586-022-05049-6</a></li>
<li>Li, Heng, and Richard Durbin. “Fast and Accurate Short Read Alignment with Burrows–Wheeler Transform.” <em>Bioinformatics</em> 25, no. 14 (July 15, 2009): 1754–60. <a href="https://doi.org/10.1093/bioinformatics/btp324">DOI: 10.1093/bioinformatics/btp324</a></li>
<li>Yousif, Mukhlid, et al. “SARS-CoV-2 Genomic Surveillance in Wastewater as a Model for Monitoring Evolution of Endemic Viruses.” <em>Nature Communications</em> 14, no. 1 (October 10, 2023): 6325. <a href="https://doi.org/10.1038/s41467-023-41369-5">DOI: 10.1038/s41467-023-41369-5</a></li>
</ul>












                
              </article>
            </div>
          
          
  <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var tab,labels=set.querySelector(".tabbed-labels");for(tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script>

<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"/></svg>
  Back to top
</button>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      Copyright (c) 2024, NICD-Wastewater-Genomics
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    <script id="__config" type="application/json">{"base": "..", "features": ["navigation.tabs", "navigation.sections", "navigation.expand", "toc.integrate", "navigation.top", "search.suggest", "search.highlight", "content.tabs.link", "content.code.annotation", "content.code.copy"], "search": "../assets/javascripts/workers/search.b8dbb3d2.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../assets/javascripts/bundle.fe8b6f2b.min.js"></script>
      
    
  </body>
</html>