
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
        <link rel="prev" href="../sequencing/">
      
      
        <link rel="next" href="../influenza/">
      
      
      <link rel="icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.0, mkdocs-material-9.5.30">
    
    
      
        <title>SARS-CoV-2 Quantitative Levels - NICD Wastewater Genomics Syndicate</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.3cba04c6.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="light-green" data-md-color-accent="purple">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#sars-cov-2-quantitative-levels-analysis" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="NICD Wastewater Genomics Syndicate" class="md-header__button md-logo" aria-label="NICD Wastewater Genomics Syndicate" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            NICD Wastewater Genomics Syndicate
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              SARS-CoV-2 Quantitative Levels
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="default" data-md-color-primary="light-green" data-md-color-accent="purple"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M17 6H7c-3.31 0-6 2.69-6 6s2.69 6 6 6h10c3.31 0 6-2.69 6-6s-2.69-6-6-6zm0 10H7c-2.21 0-4-1.79-4-4s1.79-4 4-4h10c2.21 0 4 1.79 4 4s-1.79 4-4 4zM7 9c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="slate" data-md-color-primary="light-green" data-md-color-accent="lime"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M17 7H7a5 5 0 0 0-5 5 5 5 0 0 0 5 5h10a5 5 0 0 0 5-5 5 5 0 0 0-5-5m0 8a3 3 0 0 1-3-3 3 3 0 0 1 3-3 3 3 0 0 1 3 3 3 3 0 0 1-3 3Z"/></svg>
      </label>
    
  
</form>
      
    
    
      <script>var media,input,key,value,palette=__md_get("__palette");if(palette&&palette.color){"(prefers-color-scheme)"===palette.color.media&&(media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']"),palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent"));for([key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
            
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  
  
    <li class="md-tabs__item">
      <a href=".." class="md-tabs__link">
        
  
    
  
  Home

      </a>
    </li>
  

      
        
  
  
  
    <li class="md-tabs__item">
      <a href="../sample_maps/" class="md-tabs__link">
        
  
    
  
  Sample Maps

      </a>
    </li>
  

      
        
  
  
  
    <li class="md-tabs__item">
      <a href="../labtat/" class="md-tabs__link">
        
  
    
  
  Turnaround Times

      </a>
    </li>
  

      
        
  
  
  
    <li class="md-tabs__item">
      <a href="../sequencing/" class="md-tabs__link">
        
  
    
  
  Sequencing Analysis

      </a>
    </li>
  

      
        
  
  
    
  
  
    <li class="md-tabs__item md-tabs__item--active">
      <a href="./" class="md-tabs__link">
        
  
    
  
  SARS-CoV-2 Quantitative Levels

      </a>
    </li>
  

      
        
  
  
  
    <li class="md-tabs__item">
      <a href="../influenza/" class="md-tabs__link">
        
  
    
  
  Influenza Analysis

      </a>
    </li>
  

      
    </ul>
  </div>
</nav>
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  


  

<nav class="md-nav md-nav--primary md-nav--lifted md-nav--integrated" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="NICD Wastewater Genomics Syndicate" class="md-nav__button md-logo" aria-label="NICD Wastewater Genomics Syndicate" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    NICD Wastewater Genomics Syndicate
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Home
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../sample_maps/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Sample Maps
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../labtat/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Turnaround Times
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../sequencing/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Sequencing Analysis
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  <span class="md-ellipsis">
    SARS-CoV-2 Quantitative Levels
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  <span class="md-ellipsis">
    SARS-CoV-2 Quantitative Levels
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#sars-cov-2-quantitative-levels-analysis" class="md-nav__link">
    <span class="md-ellipsis">
      SARS-CoV-2 Quantitative Levels Analysis
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#preparing-clinical-case-data" class="md-nav__link">
    <span class="md-ellipsis">
      Preparing clinical case data
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#preparing-wastewater-levels" class="md-nav__link">
    <span class="md-ellipsis">
      Preparing wastewater levels
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#plotting-clinical-cases-against-wastewater-levels" class="md-nav__link">
    <span class="md-ellipsis">
      Plotting clinical cases against wastewater levels
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#creating-a-district-level-plot" class="md-nav__link">
    <span class="md-ellipsis">
      Creating a district level plot
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../influenza/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Influenza Analysis
  </span>
  

      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


<h2 id="sars-cov-2-quantitative-levels-analysis">SARS-CoV-2 Quantitative Levels Analysis</h2>
<p>The National Institute for Communicable Diseases (NICD) currently oversees SARS-CoV-2 surveillance through two key methods: 
tracking changes in reported clinical cases documented in the NICD's Notifiable Medical Conditions Reporting's(NMC) register, 
and observing variations in SARS-CoV-2 levels detected by performing a PCR test on wastewater samples. 
These dual metrics, clinical data and wastewater analysis, can be conveniently visualized together in a unified graph. 
This guide offers detailed instructions on performing such analysis and visualization.</p>
<p>The graphs are produced using R version 4.2.2 or higher and the following packages are required to run the script:</p>
<pre><code>library (ggplot2)
library(ggthemes)
library(scales)
library(tidyverse)
library(readxl)
library(dplyr)
library(lubridate)
library(rlang)
library(writexl)
</code></pre>
<h1 id="preparing-clinical-case-data">Preparing clinical case data</h1>
<p>Access to clinical data is restricted, and permissions must be obtained from the NMC.
Once Access is obtained, clinical case data is downloaded from the NMC linelist. The data is exported as excel files. 
Due to the large volume of cases in some years, there may be multiple excel files produced, as excel has a limit in the number of
rows that per file. The first step would be import all the files into R dataframes: </p>
<pre><code>cov211&lt;- read_xlsx("/path/to/file/SARS//covid_21_1.xlsx")
cov212&lt;- read_xlsx("/path/to/file/SARS//covid_21_2.xlsx")
cov213&lt;- read_xlsx("/path/to/file/SARS//covid_21_3.xlsx")
cov214&lt;- read_xlsx("/path/to/file/SARS//covid_21_4.xlsx")
cov215&lt;- read_xlsx("/path/to/file/SARS//covid_21_5.xlsx")
cov216&lt;- read_xlsx("/path/to/file/SARS//covid_21_6.xlsx")
cov217&lt;- read_xlsx("/path/to/file/SARS//covid_21_7.xlsx")
cov218&lt;- read_xlsx("/path/to/file/SARS//covid_21_8.xlsx")
cov219&lt;- read_xlsx("/path/to/file/SARS//covid_21_9.xlsx")
cov2110&lt;- read_xlsx("/path/to/file/SARS//covid_21_10.xlsx")
cov2111&lt;- read_xlsx("/path/to/file/SARS//covid_21_11.xlsx")
cov221&lt;- read_xlsx("/path/to/file/SARS//covid_22_1.xlsx")
cov222&lt;- read_xlsx("/path/to/file/SARS//covid_22_2.xlsx")
cov223&lt;- read_xlsx("/path/to/file/SARS/covid_22_3.xlsx")
cov23&lt;- read_xlsx("/path/to/file/SARS/covid_23.xlsx")
cov24&lt;- read_xlsx("/path/to/file/SARS/covid_24.xlsx")
</code></pre>
<p>Now that we have multiple dataframes created, we want to create one dataframe to work with. To do this we are going to 
bind each of the dataframes into one "covcases" dataframe using the bind_rows command. Bind rows allows you to merge dataframe that have 
the same column names in such a way that all the rows in one dataframe are added below the rows of another. </p>
<pre><code>covcases &lt;- bind_rows(cov211, cov212, cov213, cov214, cov215, cov216, cov217, cov218, cov219, 
                  cov2110, cov2111, cov221, cov222,cov223, cov23, cov24)
</code></pre>
<p>At this point, you may want to have a look at the clinical cases in the dataframe (df) in which case you can view them using: 
    View(cases)</p>
<p>If you're unsure of all the disctricts included in the df, you can create a table to count all the unique districts </p>
<pre><code>table(covcases$District)
</code></pre>
<p>As there are many districts across the country, we only want to filter for districts in which we collect
wastewater samples. To do this we'll use the filter command from the tidyverse library. </p>
<pre><code>covcases &lt;- covcases %&gt;%
filter(District == "GP CITY OF JOHANNESBURG METRO" | District == "City of Johannesburg Metro" |
       District == "FS MANGAUNG METRO" | District == "Mangaung Metro" |
       District == "EC NELSON MANDELA BAY METRO" | District == "eThekwini Metro" |
       District == "GP CITY OF TSHWANE METRO" | District == "City of Tshwane Metro" |
       District == "GP EKURHULENI METRO" | District == "Ekurhuleni Metro" |
       District == "KZN ETHEKWINI METRO" | District == "eThekwini Metro" |
       District == "WC CITY OF CAPE TOWN METRO" | District == "City of Cape Town Metro" |
       District == "EC BUFFALO CITY METRO"| District == "Buffalo City Metro" |
       District == "NW BOJANALA PLATINUM"| District == "Bojanala Platinum" |
       District == "NW DR KENNETH KAUNDA"| District ==  "Dr Kenneth Kaunda" )
</code></pre>
<p>For most surveillance reports produced by the NICD, results are reported according to epidemiological weeks (epiweeks). 
For this report we will also be using epiweeks to report our findings. In order to convert the sample collection dates to epiweeks, 
we'll use the lubridate library to create a new column with the epiweeks of laboratory confirmed SARS-CoV-2 cases: </p>
<pre><code>covcases &lt;- covcases %&gt;% 
filter(Diagnosis_Method == "Laboratory confirmed")
covcases$newcoldate &lt;- format(as.Date(covcases$Notification_Date, format = "%Y/%m/%d"), "%Y-%m-%d") #changing the format of the date from y/m/d to ymd 
covcases$epiweek &lt;- lubridate::epiweek(ymd( covcases$newcoldate)) #generate epiweek
</code></pre>
<p>In the epiweek column you will find a number representing the epiweek the sample was collected in. For example, if a sample was collected on 2024-04-25,
then the number would be "17". However, we like to report our epiweeks with the year combined (eg 2024w17). This is just a matter of preference, 
and may be modified to suit your preferences. In order to add the year onto the epiweek with a "w" inbetween, we first create a column with the year
the sample was created, and then another column with "w" and concatenate the year, "w" and epiweek column to get our desired format:</p>
<pre><code>covcases$year &lt;- strftime(covcases$newcoldate, "%Y") #Creating year column  
covcases$week &lt;- "w" #added column with w
my_cols &lt;- c("year", "week", "epiweek") #new data object with 3 columns combined
covcases$epiweek2 &lt;- do.call(paste, c(covcases[my_cols],sep ="")) #created new variable using concat columns

cases1 &lt;- covcases
</code></pre>
<p>Since we report findings nationally and at a district level, we want to also create dataframes for each district. We'll do this by filtering for only the 
clinical cases found in that district. As R is case-sensitive, we'll filter for all variations in spelling of the districts found in the linelist. </p>
<pre><code>joburg_cases &lt;- cases1 %&gt;%
filter(District == "GP CITY OF JOHANNESBURG METRO" | District == "City of Johannesburg Metro")

mangaung_cases &lt;- cases1 %&gt;%
filter(District == "FS MANGAUNG METRO" | District == "Mangaung Metro")

nelson_cases &lt;- cases1 %&gt;%
filter(District == "EC NELSON MANDELA BAY METRO" | District == "eThekwini Metro" )

tshwane_cases &lt;- cases1 %&gt;%
filter( District == "GP CITY OF TSHWANE METRO" | District == "City of Tshwane Metro")

ekurhuleni_cases &lt;- cases1 %&gt;%
filter(District == "GP EKURHULENI METRO" | District == "Ekurhuleni Metro" )

ethekwini_cases &lt;- cases1 %&gt;%
filter(District == "KZN ETHEKWINI METRO" | District == "eThekwini Metro")

capetown_cases &lt;- cases1 %&gt;%
filter(District == "WC CITY OF CAPE TOWN METRO" | District == "City of Cape Town Metro" )

buffalo_cases &lt;- cases1 %&gt;%
filter(District == "EC BUFFALO CITY METRO"| District == "Buffalo City Metro" )

bojanala_cases &lt;- cases1 %&gt;%
filter(District == "NW BOJANALA PLATINUM"| District == "Bojanala Platinum" )

jb_marks_cases &lt;- cases1 %&gt;%
filter(District == "NW DR KENNETH KAUNDA"| District ==  "Dr Kenneth Kaunda" )
</code></pre>
<h1 id="preparing-wastewater-levels">Preparing wastewater levels</h1>
<p>There are two ways in which you can import the wastewater data in this R script. The first is by importing a csv file</p>
<pre><code>water &lt;- read.csv("path/to.file.csv")
</code></pre>
<p>Or using an API token from RedCap, which is the data management system used by our team. We usually use the API token as it allows us to 
automatically run this script without having to first download directly from RedCap. </p>
<pre><code>token &lt;- "insert your private token here"
url &lt;- "insert RedCap URL here"
formData &lt;- list("token"=token,
             content='record',
             action='export',
             format='csv',
             type='flat',
             csvDelimiter='',
             rawOrLabel='label',
             rawOrLabelHeaders='raw',
             exportCheckboxLabel='false',
             exportSurveyFields='false',
             exportDataAccessGroups='false',
             returnFormat='json'
)
response &lt;- httr::POST(url, body = formData, encode = "form")
result &lt;- httr::content(response)

water &lt;- result
</code></pre>
<p>We then filter for the wastewater treatment plants (WWTP) that we would like to include in the report: </p>
<pre><code>water &lt;- water %&gt;% 
  filter( site_name == "Northern Wastewater Treatment Works (GP)" |
        site_name == "Northern Wastewater Treatment Works (KZN)"| 
        site_name == "Rooiwal Wastewater Treatment Works"| 
        site_name ==  "ERWAT Vlakplaat Wastewater Treatment Works"|
        site_name ==  "Central Wastewater Treatment Works (KZN)"|
        site_name ==  "Goudkoppies Wastewater Treatment Works"|
        site_name ==  "Hartebeesfontein Waterworks"|
        site_name ==  "Daspoort Wastewater Treatment Works" |
        site_name ==  "Zandvleit Wastewater Treatment Works"|
        site_name == "Borcheds Quarry Wastewater Treatment Works"|
        site_name == "Brickfield Pre-treatment Works" |
        site_name == "Sterkwater Wastewater Treatment Works"|
        site_name == "Bloemspruit Wastewater Treatment Works"|
        site_name ==  "Kwanobuhle Wastewater Treatment Works" |
        site_name == "East Bank Wastewater Treatment Works"| 
        site_name ==  "Mdantsane Wastewater Treatment Works" |
        site_name == "Mmabatho Water Treatment Works" |
        site_name == "Musina WWTW (in town)" |
        site_name == "Kingstonvale"|
        site_name == "Boitekong"| 
        site_name == "Rustenburg Wastewater Treatment Works" |
        site_name == "Nancefield"| 
        site_name == "Komatipoort Sewage plant"|
        site_name == "Mahikeng Water TreatmentWorks")
</code></pre>
<p>We then generate an epiweeks column for the wastewater samples based on the sample collection dates. We also set up the columns we need
by simplifying the column names, and creating a column with the log genome copies per ml </p>
<pre><code>water1 &lt;- water
water1 &lt;- water1 %&gt;% arrange(ymd(water1$sam_col_date))
water1$epiweek &lt;- lubridate::epiweek(ymd(water1$sam_col_date)) #generate epiweek
water1$year &lt;- strftime(water1$sam_col_date, "%Y") #Creating year column  
water1$week &lt;- "w" #added column with w
my_cols &lt;- c("year", "week", "epiweek") #new data object with 3 columns combined
water1$epiweek2 &lt;- do.call(paste, c(water1[my_cols],sep ="")) #created new variable using concat columns

water1$levels &lt;- water1$n_gene_ml
water1$Date &lt;- water1$sam_col_date
water1$Result &lt;-water1$sars_cov_2_pcr_result
water1 &lt;- water1 %&gt;%
  mutate(levels = na_if(levels, levels &lt; 0))
water1 &lt;- water1 %&gt;%
  mutate(levels = if_else(levels&lt;2.34, 2.34, levels)) #replace lower than 2.34 with 2.34 (limit of quantification for qPCR)
water1$loglevels &lt;- log10(water1$levels)
</code></pre>
<p>Since wastewater levels were initially tested using a qPCR and later change to dPCR, we add a quality check for the "PCR" column which ensure the correct 
type of PCR is indicated based on when we switched to dPCR (epiweek 30 of 2023) </p>
<pre><code>water1 &lt;- water1 %&gt;%
  mutate(PCR = if_else(year == 2021, "qPCR", PCR)) %&gt;%
  mutate(PCR = if_else(year == 2022, "qPCR", PCR)) %&gt;%
  mutate(PCR = if_else(year == 2023 &amp; week &lt; 30, "qPCR", PCR)) %&gt;%
  mutate(PCR = if_else(year == 2023 &amp; week &gt; 30, "dPCR", PCR)) %&gt;%
  mutate(PCR = if_else(year == 2024, "dPCR", PCR))
</code></pre>
<h1 id="plotting-clinical-cases-against-wastewater-levels">Plotting clinical cases against wastewater levels</h1>
<p>We'll start by plotting the national waterwater levels against the clinical cases. The wasterwater levels will be in the form of a line graph, 
and bars will be used to represent the number of clinical cases.  </p>
<p>We start by tabulating the number of samples received </p>
<pre><code>rsa_samples &lt;- cases1 %&gt;%
  group_by(epiweek2)%&gt;%
count(epiweek2, na.rm=TRUE)

rsa_water &lt;- water1 %&gt;% 
  filter( Site.Name. == "Goudkoppies Wastewater Treatment Works" |
        Site.Name. == "Northern Wastewater Treatment Works"| 
        Site.Name. == "Rooiwal Wastewater Treatment Works" |
        Site.Name. == "Daspoort Wastewater Treatment Works" |
        Site.Name. == "ERWAT Vlakplaat Wastewater Treatment Works" |
        Site.Name. == "Hartebeesfontein Waterworks" |
        Site.Name. == "Zandvleit Wastewater Treatment Works" |
        Site.Name. == "Borcheds Quarry Wastewater Treatment Works" |
        Site.Name. == "East Bank Wastewater Treatment Works " |
        Site.Name. == "Mdantsane Wastewater Treatment Works"|
        Site.Name. == "Brickfield Pre-treatment Works  " |
        Site.Name. == "Kwanobuhle Wastewater Treatment Works"|
        Site.Name. == "Sterkwater Wastewater Treatment Works " |
        Site.Name. == "Bloemspruit Wastewater Treatment Works"| 
        Site.Name. == "Central Wastewater Treatment Works " |
        Site.Name. == "Northern Wastewater Treatment Works"|
        Site.Name. == "Mahikeng Water Treatment Works"|
        Site.Name. == "Mmabatho Water Treatment Work"|
        Site.Name. == "Rustenburg Wastewater Treatment Works")

rsacopies &lt;- rsa_water %&gt;% 
  group_by(epiweek2)%&gt;%
  summarise(sum_genomes = sum(levels,na.rm = TRUE),
        .groups = 'keep')

rsacopies["sum_genomes"][rsacopies["sum_genomes"] == 0] &lt;- NA

rsacases_vs_water&lt;- full_join(rsa_samples, rsa_water, by= "epiweek2")
rsacases_vs_water&lt;- full_join(rsacases_vs_water, rsacopies, by= "epiweek2")
</code></pre>
<p>We then repeat this for weeks where there are clinical cases but no wastewater samples otherwise it would be blank</p>
<pre><code>rsacases_vs_water &lt;- rsacases_vs_water %&gt;%
  mutate(PCR = if_else(year == 2021, "qPCR", PCR)) %&gt;%
  mutate(PCR = if_else(year == 2022, "qPCR", PCR)) %&gt;%
  mutate(PCR = if_else(year == 2023 &amp; week &lt; 30, "qPCR", PCR)) %&gt;%
  mutate(PCR = if_else(year == 2023 &amp; week &gt; 30, "dPCR", PCR)) %&gt;%
  mutate(PCR = if_else(year == 2024, "dPCR", PCR)) %&gt;%
  mutate(PCR = if_else(is.na(PCR), "qPCR", PCR))


rsacases_vs_water$final_result &lt;- rsacases_vs_water$SARS.CoV.2.PCR.result
</code></pre>
<p>Since we don't need all the columns currently in the dataframe, select the columns we want, which is a bit easier to work with.
    rsacases_vs_water &lt;- rsacases_vs_water %&gt;%
      select(epiweek2, n, sum_genomes, Date, PCR)</p>
<p>We calculate log10 of sum_genomes and store it in a new column loglevels</p>
<pre><code>rsacases_vs_water$loglevels &lt;- log10(rsacases_vs_water$sum_genomes)
</code></pre>
<p>In our reports we use square markers to indicate weeks where samples were tested. We place these markers below zero on the x-axis, in line with the corresponding epiweek. In order to do this we create a new column (tested). 
We use the mutate function to create these columns and assign the value -0.3 if a sample was tested. 
The values -0.3 is arbitrary, and was selected to ensure the marker is graphed below zero.</p>
<pre><code>rsacases_vs_water&lt;- rsacases_vs_water %&gt;%
  mutate(tested = case_when( (sum_genomes &gt; 0) ~ -0.3)) %&gt;%
  mutate(n = if_else(is.na(n), 0, n)) #replace na with 0
</code></pre>
<p>Before plotting, we want to ensure that the correct order of the epiweeks such that the order is by year first, then week. To do this we first create a copy of epiweek2 and name it epiweek3</p>
<pre><code>rsacases_vs_water$epiweek3 &lt;- rsacases_vs_water$epiweek2
</code></pre>
<p>Then we split the epiweek3 column into a "year" column and a "week" column. We then convert the values in these columns into integers.</p>
<pre><code>rsacases_vs_water &lt;- rsacases_vs_water %&gt;%
  separate(epiweek3, sep = "w", into = c("year", "week")) %&gt;%
  mutate(across(c("year", "week"), as.integer))
</code></pre>
<p>Since we only started testing wastewater samples for SARS-CoV-2 in 2021, we filter out and clinical samples from 2020</p>
<pre><code>rsacases_vs_water &lt;- rsacases_vs_water %&gt;%
  filter(year != 2020)
</code></pre>
<p>Now we can order the samples first by year and then week</p>
<pre><code>rsacases_vs_water &lt;-  rsacases_vs_water[ #ordering by year first then week
  with(rsacases_vs_water, order(year, week)),
]
</code></pre>
<p>To ensure this order stays the same when we're plotting the graph, we conver epiweek2 into an ordered factor.</p>
<pre><code>rsacases_vs_water$epiweek2 &lt;- factor(rsacases_vs_water$epiweek2, levels = unique(rsacases_vs_water$epiweek2), ordered = T)
</code></pre>
<p>Add a new column "Country" with a constant value.</p>
<pre><code>rsacases_vs_water$Country &lt;- "South African SARS-CoV-2 Wastewater Levels"

rsacases_vs_water$Date &lt;- "w"
</code></pre>
<p>Remove duplicated rows </p>
<pre><code>rsacases_vs_water &lt;- rsacases_vs_water %&gt;% distinct()
</code></pre>
<p>Create a new column 'end' by parsing date and time using the lubridate package. Then we concatenate year, week, and 0 with "-" as separator to form a string.
Specify the format 'Y-W-w' where 'Y' is year, 'W' is week, and 'w' is the day of the week (0 = week start on Sunday).Add 6 days to the parsed date to get the end of the epidemiological week.
This function parses the concatenated string into a date-time object.The resulting date represents the start of the epidemiological week, so adding 6 days gives the end of the week.
This is mainly used for getting the last day of the epidemilogical week which is used on the dashboard. </p>
<pre><code>rsacases_vs_water$end &lt;- lubridate::parse_date_time(paste(rsacases_vs_water$year, rsacases_vs_water$week,0, sep="-"),'Y-W-w') + days(6)
#lubridate::parse_date_time(year, week, 0= week start on sunday, sep = formwat you want)
#this gives start of epiweek so add 6 days to get end of epiweek
</code></pre>
<p>This next line of code is only necessary if you have many samples spanning multiple years. You may want to only show every nth epiweek on the graph instead of every week as this may appear cluttered. 
This funtion allows you to return every nth element. When added to the graph, you specify the value of n.</p>
<pre><code>every_nth = function(n) {
  return(function(x) {x[c(TRUE, rep(FALSE, n - 1))]})
}
</code></pre>
<p>We then plot the graph.This line opens a PNG device to create a file named overall_rsa.png.
The dimensions are set to 5 times 950 pixels in width and 5 times 300 pixels in height.
The resolution is set to 300 DPI.
The point size for text and points in the plot is set to 8. </p>
<pre><code>png("/path/to/file/overall_rsa.png",  # Create a PNG file for the plot
    width = 5 * 950,  # Set width of the image
    height = 5 * 300,  # Set height of the image
    res = 300,  # Set resolution of the image
    pointsize = 8)  # Set point size for the image
</code></pre>
<p>This initializes a ggplot object using the rsacases_vs_water data frame.</p>
<pre><code>rsaplot &lt;- ggplot(rsacases_vs_water) +  # Create a ggplot object with rsacases_vs_water data frame
</code></pre>
<p>This adds points to the plot representing the tested samples multiplied by 10000.
Points are grouped by 1 and colored according to "Sample Collection".
The size of the points is set to 1, and the shape is set to 15.</p>
<pre><code>geom_point(aes(x = epiweek2, y = tested * 10000, group = 1, col = "Sample Collection"), stat = "identity", size = 1, shape = 15) +  # Add points for sample collection
</code></pre>
<p>This adds a bar plot for the number of samples (n) for each epidemiological week (epiweek2).
Bars are filled and colored gray.</p>
<pre><code>geom_bar(aes(x = epiweek2, y = n), stat = "identity", fill = "gray", colour = "gray") +  # Add bar plot for number of cases
</code></pre>
<p>This adds points for log genome copies multiplied by 10000.
Points are grouped and colored by the Country variable.
The size of the points is set to 2.</p>
<pre><code>geom_point(aes(x = epiweek2, y = loglevels * 10000, group = Country, col = Country), stat = "identity", size = 2) +  # Add points for log genome copies
</code></pre>
<p>This adds lines for log genome copies multiplied by 10000.
Lines are grouped and colored by the Country variable.
The size of the lines is set to 1.</p>
<pre><code>geom_line(aes(x = epiweek2, y = loglevels * 10000, group = Country, col = Country), stat = "identity", size = 1) +  # Add lines for log genome copies
</code></pre>
<p>This sets the x-axis breaks to show every 2nd label</p>
<pre><code>scale_x_discrete(breaks = every_nth(n = 2)) +  # Set x-axis breaks to show every 2nd label
</code></pre>
<p>This manually sets the colors for the points and lines.
"Sample Collection" is colored dark blue.
"South African SARS-CoV-2 Wastewater Levels" is colored #009ADE (a shade of blue).</p>
<pre><code>scale_colour_manual(values = c("Sample Collection" = "darkblue", "South African SARS-CoV-2 Wastewater Levels" = "#009ADE")) +  # Set custom colors for the points and lines
</code></pre>
<p>This customizes the legend, overriding the aesthetics for shape and line type.</p>
<pre><code>guides(color = guide_legend(override.aes = list(shape = c(15, 16), linetype = c(0, 1)))) +  # Customize legend
</code></pre>
<p>This sets the y-axis to have a secondary axis for log genome copies divided by 10000.
The secondary axis is labeled "Log Genome Copies/ml (N Gene)".
The y-axis has 4 breaks, and labels are formatted with comma</p>
<pre><code>scale_y_continuous(sec.axis = sec_axis(~ . / 10000, name = "Log Genome Copies/ml (N Gene)\n"),  # Create secondary y-axis for log genome copies
                 breaks = scales::pretty_breaks(n = 4),  # Set breaks for the y-axis
                 labels = scales::label_comma()) +  # Format y-axis labels with commas
</code></pre>
<p>This adds labels to the x-axis ("Epidemiological week") and y-axis ("Laboratory confirmed cases").</p>
<pre><code>labs(x = "\nEpidemiological week", y = "Laboratory confirmed cases\n") +  # Add axis labels
</code></pre>
<p>This creates facets for the PCR variable, with levels "qPCR" and "dPCR".
The x-axis scales are free within each facet, and the spacing is free.</p>
<pre><code>facet_grid(~factor(PCR, levels = c('qPCR', 'dPCR')), scales = "free_x", space = "free_x") +  # Create facets for PCR method with free scales
</code></pre>
<p>This customizes various theme elements:
x-axis text is rotated 90 degrees, aligned horizontally at 0, colored black, and sized 12.
y-axis text is colored black and sized 12.
The legend is positioned at the bottom, and its title is removed.
General text is colored black and sized 12.
x-axis and y-axis lines are colored black and sized 1.
Facet background is white, and facet text is sized 12</p>
<pre><code>ggthemes::theme_tufte() +  # Apply Tufte theme
theme(
    axis.text.x = element_text(angle = 90, hjust = 0, color = "black", size = 12),  # Customize x-axis text
    axis.text.y = element_text(color = "black", size = 12),  # Customize y-axis text
    legend.position = "bottom",  # Place legend at the bottom
    legend.title = element_blank(),  # Remove legend title
    text = element_text(color = "black", size = 12),  # Set general text properties
    axis.line.x = element_line(color = "black", size = 1),  # Customize x-axis line
    axis.line.y = element_line(color = "black", size = 1),  # Customize y-axis line
    strip.background = element_rect(fill = "white"),  # Customize facet background
    strip.text = element_text(size = 12)  # Customize facet text
)
</code></pre>
<p>This line displays the ggplot object created.</p>
<pre><code>rsaplot  # Plot the ggplot object
</code></pre>
<p>This closes the PNG device, saving the plot to the file.</p>
<pre><code>dev.off()  # Close the PNG device
</code></pre>
<p><img alt="Screenshot" src="../img/overall_rsa.png" /></p>
<h1 id="creating-a-district-level-plot">Creating a district level plot</h1>
<p>Below is the continuation and completion of the detailed guide for performing SARS-CoV-2 quantitative levels analysis using R. 
This guide includes plotting the data at a district level. In the example below we have selected Johannesburg as our district of interest. For other districts, replace the District Name and Site Names for 
your district of interest. </p>
<p>First start by tabulating the number of samples received. We do this by grouping the samples by epidemiological week and then counting the number of samples per week. </p>
<pre><code>jhb_samples &lt;- joburg_cases %&gt;%
  group_by(epiweek2)%&gt;%
  count(epiweek2, na.rm=TRUE)
</code></pre>
<p>Then we filter for samples that were collected in our district of interest and at the wastewater treatment plants of interest. In this case we're filtering for samples 
collected in the Johannesburg Municipality, at the Goudkoppies and Northern Wastewater Treatment Plants.  </p>
<pre><code>jhb_water &lt;- water1 %&gt;% 
  filter(District.Name == "Johannesburg MM") %&gt;%
  filter( Site.Name. == "Goudkoppies Wastewater Treatment Works" |
        Site.Name. == "Northern Wastewater Treatment Works (GP)")
</code></pre>
<p>We then merge the two data frames and create a new column for the final PCR results. </p>
<pre><code>jhbcases_vs_water&lt;- full_join(jhb_samples, jhb_water, by= "epiweek2")
jhbcases_vs_water$final_result &lt;- jhbcases_vs_water$SARS.CoV.2.PCR.result
</code></pre>
<p>We then select relevant columns we would like to work with, and remove rows that have invalid epiweek values i.e all samples in which epiweek2 = NAwNA as these are 
samples which had no sample collection dates.</p>
<pre><code>jhbcases_vs_water &lt;- jhbcases_vs_water %&gt;% 
  select(epiweek2, n, Site.Name., 
     Site.Province., District.Name,Genome.copies.mL...N.gene.., levels, loglevels, Date, final_result, PCR) %&gt;% 
  filter(epiweek2 != "NAwNA")
</code></pre>
<p>In our reports we use square markers to indicate weeks where samples were tested. We place these markers below zero on the x-axis, in line with the corresponding
epiweek. In order to do this we create new columns (tested1 and tested2) for each WWTP. We use the mutate function to create these columns and assign the value -0.3 or -0.1 for each site
if a sample was tested and had a Positive or Negative result. The values -0.3 and -0.1 are arbitrary, and were selected to ensure the marker is graphed below zero.  </p>
<pre><code>jhbcases_vs_water&lt;- jhbcases_vs_water %&gt;%
  mutate(tested1 = case_when( (Site.Name. == "Goudkoppies Wastewater Treatment Works" &amp; final_result == "Positive") ~ -0.3, 
                          (Site.Name. == "Goudkoppies Wastewater Treatment Works" &amp; final_result == "Negative") ~ -0.3)) %&gt;%
  mutate(tested2 = case_when( (Site.Name. == "Northern Wastewater Treatment Works (GP)" &amp; final_result == "Positive") ~ -0.1, 
                          (Site.Name. == "Northern Wastewater Treatment Works (GP)" &amp; final_result == "Negative") ~ -0.1))
</code></pre>
<p>Before plotting, we want to ensure that the correct order of the epiweeks such that the order is by year first, then week.  To do this we first create a copy of epiweek2 and name it epiweek3 </p>
<pre><code>jhbcases_vs_water$epiweek3 &lt;- jhbcases_vs_water$epiweek2
</code></pre>
<p>Then we split the epiweek3 column into a "year" column and a "week" column. We then convert the values in these columns into integers. </p>
<pre><code>jhbcases_vs_water &lt;- jhbcases_vs_water %&gt;%
  separate(epiweek3, sep = "w", into = c("year", "week")) %&gt;%
  mutate(across(c("year", "week"), as.integer))
</code></pre>
<p>Since we only started testing wastewater samples for SARS-CoV-2 in 2021, we filter out and clinical samples from 2020</p>
<pre><code>jhbcases_vs_water &lt;- jhbcases_vs_water %&gt;%
  filter(year != 2020)
</code></pre>
<p>Now we can order the samples first by year and then week</p>
<pre><code>jhbcases_vs_water &lt;-  jhbcases_vs_water[ #ordering by year first then week
  with(jhbcases_vs_water, order(year, week)),
]
</code></pre>
<p>To ensure this order stays the same when we're plotting the graph, we conver epiweek2 into an ordered factor.</p>
<pre><code>jhbcases_vs_water$epiweek2 &lt;- factor(jhbcases_vs_water$epiweek2, levels = unique(jhbcases_vs_water$epiweek2), ordered = T)
</code></pre>
<p>Initially samples were tested using a qPCR but during 2023, we switched to testing samples using a dPCR. This change is indicated in our graphs using facets which show the type of PCR test performed. To create
these facets, we first specify which type of PCR test was performed based on when the samples were collected. </p>
<pre><code>jhbcases_vs_water &lt;- jhbcases_vs_water %&gt;%
  mutate(PCR = if_else(year == 2021, "qPCR", PCR)) %&gt;% # Set PCR method for 2021 to qPCR
  mutate(PCR = if_else(year == 2022, "qPCR", PCR)) %&gt;% # Set PCR method for 2022 to qPCR
  mutate(PCR = if_else(year == 2023 &amp; week &lt; 30, "qPCR", PCR)) %&gt;% # Set PCR method for early 2023 to qPCR
  mutate(PCR = if_else(year == 2023 &amp; week &gt; 30, "dPCR", PCR)) %&gt;% # Set PCR method for late 2023 to dPCR
  mutate(PCR = if_else(is.na(PCR), "qPCR", PCR)) %&gt;% # Replace NA PCR values with qPCR
  mutate(n = if_else(is.na(n), 0, n)) #replace na with 0
</code></pre>
<p>This next line of code is only necessary if you have many samples spanning multiple years. You may want to only show every nth epiweek on the graph instead of every week as this may appear cluttered. 
This funtion allows you to return every nth element. When added to the graph, you specify the value of n.</p>
<pre><code>every_nth = function(n) {
  return(function(x) {x[c(TRUE, rep(FALSE, n - 1))]})
}
</code></pre>
<p>Then we create the graph and save it as a png file.</p>
<pre><code>png("/path/yo/file/Johannesburg.png", # Create a PNG file for the plot 
    width = 5*950,  # Set width of the image
    height = 5*300, # Set height of the image
    res = 300, # Set resolution of the image
    pointsize = 8) # Set point size for the image

jhbplot &lt;- ggplot(jhbcases_vs_water) +    # Create ggplot object with jhbcases_vs_water data frame
geom_bar(aes(x = epiweek2, y = n), stat = "identity", fill = "gray", colour = "gray") +   # Add bar plot for number of cases
geom_point(aes(x = epiweek2, y = tested1 * 10000, group = 1, col = "Goudkoppies Sample Collection"), stat = "identity", size = 1, shape = 15) +   # Add points for Goudkoppies sample collection
geom_point(aes(x = epiweek2, y = tested2 * 10000, group = 1, col = "Northern Sample Collection"), stat = "identity", size = 1, shape = 15) +   # Add points for Northern sample collection
geom_point(aes(x = epiweek2, y = loglevels * 10000, group = Site.Name., col = Site.Name.), stat = "identity", size = 2) +   # Add points for log genome copies
geom_line(aes(x = epiweek2, y = loglevels * 10000, group = Site.Name., col = Site.Name.), stat = "identity", size = 1) +   # Add lines for log genome copies
scale_colour_manual(values = c("Goudkoppies Sample Collection" = "darkred", "Northern Sample Collection" = "darkblue", 
                            "Goudkoppies Wastewater Treatment Works" = "#FC4E2A", "Northern Wastewater Treatment Works (GP)" = "#009ADE")) +   # Set custom colors for the points and lines
scale_y_continuous(sec.axis = sec_axis(~ . / 10000, name = "Log Genome Copies/ml (N Gene)\n"),   # Create secondary y-axis for log genome copies
                 breaks = scales::pretty_breaks(n = 2),   # Set breaks for the y-axis
                 labels = scales::label_comma()) +   # Format y-axis labels with commas
guides(color = guide_legend(override.aes = list(shape = c(15, 16, 15, 16),   # Customize legend
                                              linetype = c(0, 1, 0, 1)))) +   
scale_x_discrete(breaks = every_nth(n = 5)) +   # Set x-axis breaks to show every 5th label
facet_grid(~factor(PCR, levels = c('qPCR', 'dPCR')), scales = "free_x", space = "free") +   # Create facets for PCR method with free scales
labs(x = "\nEpidemiological week", y = "Laboratory confirmed cases\n") +   # Add axis labels
ggthemes::theme_tufte() +   # Apply Tufte theme
theme(
    axis.text.x = element_text(angle = 90, hjust = 0, color = "black", size = 16),   # Customize x-axis text
    axis.text.y = element_text(color = "black", size = 16),   # Customize y-axis text
    legend.position = "bottom",   # Place legend at the bottom
    legend.title = element_blank(),   # Remove legend title
    text = element_text(color = "black", size = 16),   # Set general text properties
    axis.line.x = element_line(color = "black", size = 1),   # Customize x-axis line
    axis.line.y = element_line(color = "black", size = 1),   # Customize y-axis line
    strip.background = element_rect(fill = "white"),   # Customize facet background
    strip.text = element_text(size = 12))   # Customize facet text

jhbplot   # Plot the ggplot object

dev.off()   # Close the PNG device
</code></pre>
<p><img alt="Screenshot" src="../img/Johannesburg.png" /></p>












                
              </article>
            </div>
          
          
  <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var tab,labels=set.querySelector(".tabbed-labels");for(tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script>

<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"/></svg>
  Back to top
</button>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      Copyright (c) 2024, NICD-Wastewater-Genomics
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    <script id="__config" type="application/json">{"base": "..", "features": ["navigation.tabs", "navigation.sections", "navigation.expand", "toc.integrate", "navigation.top", "search.suggest", "search.highlight", "content.tabs.link", "content.code.annotation", "content.code.copy"], "search": "../assets/javascripts/workers/search.b8dbb3d2.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../assets/javascripts/bundle.fe8b6f2b.min.js"></script>
      
    
  </body>
</html>