
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
        <link rel="prev" href="../sample_maps/">
      
      
        <link rel="next" href="../sequencing/">
      
      
      <link rel="icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.0, mkdocs-material-9.5.30">
    
    
      
        <title>Turnaround Times - NICD Wastewater Genomics Syndicate</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.3cba04c6.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="light-green" data-md-color-accent="purple">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#laboratory-and-sequencing-turnaround-times" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="NICD Wastewater Genomics Syndicate" class="md-header__button md-logo" aria-label="NICD Wastewater Genomics Syndicate" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            NICD Wastewater Genomics Syndicate
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Turnaround Times
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="default" data-md-color-primary="light-green" data-md-color-accent="purple"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M17 6H7c-3.31 0-6 2.69-6 6s2.69 6 6 6h10c3.31 0 6-2.69 6-6s-2.69-6-6-6zm0 10H7c-2.21 0-4-1.79-4-4s1.79-4 4-4h10c2.21 0 4 1.79 4 4s-1.79 4-4 4zM7 9c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="slate" data-md-color-primary="light-green" data-md-color-accent="lime"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M17 7H7a5 5 0 0 0-5 5 5 5 0 0 0 5 5h10a5 5 0 0 0 5-5 5 5 0 0 0-5-5m0 8a3 3 0 0 1-3-3 3 3 0 0 1 3-3 3 3 0 0 1 3 3 3 3 0 0 1-3 3Z"/></svg>
      </label>
    
  
</form>
      
    
    
      <script>var media,input,key,value,palette=__md_get("__palette");if(palette&&palette.color){"(prefers-color-scheme)"===palette.color.media&&(media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']"),palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent"));for([key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
            
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  
  
    <li class="md-tabs__item">
      <a href=".." class="md-tabs__link">
        
  
    
  
  Home

      </a>
    </li>
  

      
        
  
  
  
    <li class="md-tabs__item">
      <a href="../sample_maps/" class="md-tabs__link">
        
  
    
  
  Sample Maps

      </a>
    </li>
  

      
        
  
  
    
  
  
    <li class="md-tabs__item md-tabs__item--active">
      <a href="./" class="md-tabs__link">
        
  
    
  
  Turnaround Times

      </a>
    </li>
  

      
        
  
  
  
    <li class="md-tabs__item">
      <a href="../sequencing/" class="md-tabs__link">
        
  
    
  
  Sequencing Analysis

      </a>
    </li>
  

      
        
  
  
  
    <li class="md-tabs__item">
      <a href="../quantitative/" class="md-tabs__link">
        
  
    
  
  SARS-CoV-2 Quantitative Levels

      </a>
    </li>
  

      
        
  
  
  
    <li class="md-tabs__item">
      <a href="../influenza/" class="md-tabs__link">
        
  
    
  
  Influenza Analysis

      </a>
    </li>
  

      
    </ul>
  </div>
</nav>
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  


  

<nav class="md-nav md-nav--primary md-nav--lifted md-nav--integrated" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="NICD Wastewater Genomics Syndicate" class="md-nav__button md-logo" aria-label="NICD Wastewater Genomics Syndicate" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    NICD Wastewater Genomics Syndicate
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Home
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../sample_maps/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Sample Maps
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  <span class="md-ellipsis">
    Turnaround Times
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  <span class="md-ellipsis">
    Turnaround Times
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#laboratory-and-sequencing-turnaround-times" class="md-nav__link">
    <span class="md-ellipsis">
      Laboratory And Sequencing Turnaround Times
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#setting-up" class="md-nav__link">
    <span class="md-ellipsis">
      Setting up
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Setting up">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#calculating-the-epidemiological-weeks-epiweeks" class="md-nav__link">
    <span class="md-ellipsis">
      Calculating the epidemiological weeks (epiweeks)
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#calculating-turnaround-times-for-quantitative-results" class="md-nav__link">
    <span class="md-ellipsis">
      Calculating turnaround times for quantitative results
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#calculating-turnaround-times-for-sequencing-results" class="md-nav__link">
    <span class="md-ellipsis">
      Calculating turnaround times for sequencing results
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../sequencing/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Sequencing Analysis
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../quantitative/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    SARS-CoV-2 Quantitative Levels
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../influenza/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Influenza Analysis
  </span>
  

      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


<h2 id="laboratory-and-sequencing-turnaround-times">Laboratory And Sequencing Turnaround Times</h2>
<p>Timely reporting of sample test results is required to support the wastewater-based 
surveillance of pathogens. This ensure that any changes in the transmission of pathogens circulating 
within the population or even circulating variants may be detected in a timely manner. </p>
<p>Laboratory turnaround times (tat) are plotted each week to help rapidly identify delays at various 
points at which samples are processed. This helps us quickly identify what may be causing delays and 
address any challenges which may be faced. </p>
<p>There are two graphs generated: one for the time taken to produce the quantitative levels of SARS-CoV-2 in wastewater, 
and the other for the time taken to produce the sequencing results for those samples. The graphs produced illustrate the
number of samples collected and processed during each week as a bar graph, and a line graph representing the average turnaround 
time of all samples during a given week. </p>
<h1 id="setting-up">Setting up</h1>
<p>The graphs are produced using R version 4.2.2 or higher and the following packages are required to run the script: </p>
<pre><code>library(ggthemes)
library(scales)
library(tidyverse)
library(readxl)
library(lubridate)
</code></pre>
<p>RedCap is used by our team as a database to capture and store sample information. There are two ways in which data may be exported for use . The first
is to download a csv file from RedCap. The read.csv() function will be used to import the the csv file into
a dataframe. The types of data within each column may be specified using col_types. The coloumns don't always have to specied,but sometimes dates may be imported as numeric values. So specifying the columns helps prevent that from happening. </p>
<pre><code>water &lt;- read.csv("Path/to/the/file.csv", check.names = F) #Replace the path with the path to the file on your computer
</code></pre>
<p>Alternatively, the RedCap API token may be used to automatically connect to RedCap within the R script, such that fresh data is taken from RedCap without the need to manually export the data. 
To obtain the API token, log into RedCap and select API on the left hand side under the Applications tab. Generate an API token and insert into the code below (Ensure you have permissions from your administrator). The RedCap API 
playground may also be used if you would like to modify the code below to suit your specific requirements.</p>
<pre><code>token &lt;- "insert-api-toke-here"
url &lt;- "insert-redcap-url-here"
formData &lt;- list("token"=token,
             content='record',
             action='export',
             format='csv',
             type='flat',
             csvDelimiter='',
             rawOrLabel='raw',
             rawOrLabelHeaders='raw',
             exportCheckboxLabel='false',
             exportSurveyFields='false',
             exportDataAccessGroups='false',
             returnFormat='json'
)
response &lt;- httr::POST(url, body = formData, encode = "form")
result &lt;- httr::content(response)

water &lt;- result
</code></pre>
<p>Before getting started, we're going to ensure that all date columns are being read in by R as a date. This step is not really necessary, but good to have. </p>
<pre><code>water$sam_col_date &lt;- as.Date(water$sam_col_date)
water$proc_date &lt;- as.Date(water$proc_date)
water$pcr_test_date &lt;- as.Date(water$pcr_test_date)
water$extraction_date &lt;- as.Date(water$extraction_date )
water$pcr_date &lt;- as.Date(water$pcr_date)
water$tapestation_date &lt;- as.Date(water$tapestation_date )
water$date_rec_sequences &lt;- as.Date(water$date_rec_sequences)
water$date_sub_sequencing &lt;- as.Date(water$date_sub_sequencing)
water$seq_report_date &lt;- as.Date(water$seq_report_date)
</code></pre>
<h2 id="calculating-the-epidemiological-weeks-epiweeks">Calculating the epidemiological weeks (epiweeks)</h2>
<p>We now need to add in a column containing the epiweeks. To ensure that no errors occur in our original "water" dataframe while we perform some data wrangling,
we'll create a second dataframe, which we'll call "water1"</p>
<pre><code>water1&lt;- water
</code></pre>
<p>The strftime() function can be used to create a year and month column based on when the sample was collected </p>
<pre><code>water1$year &lt;- strftime(water1$`Date sample collected`, "%Y") #Creating year column  
water1$month &lt;- strftime(water1$`Date sample collected`, "%m") #Creating month column
</code></pre>
<p>The epiweek may also be calculated using the lubridate to calculate the epiweek based on the date that the sample was collected.
Lubricate requires that the date format be specified and since our dates are recorded as yyyy-mm-dd we'll specify that the formwat is ymd</p>
<pre><code>water1$epiweek &lt;- lubridate::epiweek(ymd(water1$`Date sample collected`))
</code></pre>
<p>For most our reports we report the epiweeks with the year combined (e.g. 2023w45)
We can create a column containing the combined year and epiweek. We start by creating a column with the week represented as a "w" </p>
<pre><code>water1$week &lt;- "w" #added column with w
</code></pre>
<p>Week can then combine what's in the "year", "week" and "epiweek" column and then send it to a new variable called "epiweek2"</p>
<pre><code>my_cols2 &lt;- c("year", "week", "epiweek") #new data object with 3 columns combined
water1$epiweek2 &lt;- do.call(paste, c(water1[my_cols2],sep ="")) #created new variable using concat columns
</code></pre>
<p>To ensure that all data is ordered by the year the sample was collected, and then by epidemiological week </p>
<pre><code>water &lt;-  water[ #ordering by year first then week
with(water, order(year, week)),
]
</code></pre>
<h1 id="calculating-turnaround-times-for-quantitative-results">Calculating turnaround times for quantitative results</h1>
<p>Because we only started recording the tat around the end of April 2023, we have to filter out all other samples collected before this period </p>
<pre><code>watertat &lt;- water1 %&gt;% 
filter(sam_col_date &gt; "2023-04-30")
</code></pre>
<p>The tat for each sample may then be calculated by calculating the difference in time by various processes conducted in the lab. The difference is calculated in days</p>
<pre><code>watertat$pro_tat &lt;- difftime(watertat$`Date Processed`,watertat$`Date sample collected`, units = c("days"))
watertat$lab_tat &lt;- difftime(watertat$`Date tested at Laboratory`,watertat$`Date Processed`, units = c("days"))
watertat$overall_tat &lt;- difftime(watertat$`Date tested at Laboratory`,watertat$`Date sample collected`, units = c("days"))
</code></pre>
<p>To calculate the mean tat for all samples given a specific epiweek </p>
<pre><code>mean_pro &lt;- watertat%&gt;% # we specify which df we would like to use
group_by(epiweek2)%&gt;% # we specify what we would like to group by 
summarise(Meanpro=mean(pro_tat, na.rm = TRUE))

mean_lab &lt;- watertat%&gt;% # we specify which df we would like to use
group_by(epiweek2)%&gt;% # we specify what we would like to group by 
summarise(Meanlab=mean(lab_tat, na.rm = TRUE))

mean_overall &lt;- watertat%&gt;% # we specify which df we would like to use
group_by(epiweek2)%&gt;% # we specify what we would like to group by 
summarise(Meanoverall=mean(overall_tat, na.rm = TRUE))
</code></pre>
<p>Since we've calculated the tat, we need to created a new dataframe with the 
mean tat and the number of samples collected during a given epiweek. To tabulate the number of samples collected during each epiweek we can do </p>
<pre><code>freq_samples &lt;- watertat %&gt;%
group_by(epiweek2)%&gt;%
count(epiweek2, na.rm=TRUE)
</code></pre>
<p>We then need create a new dataframe which contains the number of samples collected during each week
and  how long it took to process those samples</p>
<pre><code>tat_vs_freq &lt;- merge(mean_pro, mean_lab, by= "epiweek2", na.rm =TRUE)
tat_vs_freq &lt;- merge(tat_vs_freq, mean_overall, by= "epiweek2", na.rm =TRUE)
tat_vs_freq &lt;- merge(tat_vs_freq, freq_samples, by= "epiweek2", na.rm =TRUE)
</code></pre>
<p>Since we've captured our epiweeks in the year-w-week format (eg. 2023w45), when plotting this on the x-axis, R will plot the epiweeks in lexicographic order
instead of by year first and then by week. In order to force R to plot the epiweeks in the correct order, we will first creat a new column called epiweek 3. Epiweek 3 will be 
a duplication of epiweek 2. We will then separate the year and week and  order the dataframe by year and then week.</p>
<pre><code>tat_vs_freq$epiweek3 &lt;- tat_vs_freq$epiweek2

tat_vs_freq &lt;- tat_vs_freq %&gt;% 
    separate(epiweek3,into=c("year", "week"), 
     sep="w", convert = TRUE, extra = "merge")

tat_vs_freq &lt;-  tat_vs_freq[ #ordering by year first then week
with(tat_vs_freq, order(year, week)),
]
</code></pre>
<p>Since we will be plotting from epiweek2, we now tell R to treat epiweek2 as a factor, and that if there are multiple entries for a given epiweek it must only take one unique epiweek. Lastly, that it must keep the same order
as the dataframe</p>
<pre><code>tat_vs_freq$epiweek2 &lt;- factor(tat_vs_freq$epiweek2, levels = unique(tat_vs_freq$epiweek2), ordered = T)
</code></pre>
<p>Now that we have a dataframe (tat_vs_freq) that contains how many samples were collected during each epiweek and how long it took to process those samples, 
we can plot that on a graph. </p>
<p>To specify that the graph should be saved as a .png file, we can use the png() function and specify the path of the folder
for which the file should be saved in. The dimensions of the graph may also be specified to your desired width and height. </p>
<pre><code>png("path/to/file/quantitative.png", 
width = 5*900,
height = 5*500,
res = 300,
pointsize = 8)
</code></pre>
<p>To plot the graph, ggplot2 will be used. We first specify which dataframe will be used to build the graph:</p>
<pre><code>tat_plot &lt;- ggplot(tat_vs_freq) +
</code></pre>
<p>To specify that we want to create a bar dark gray bar graph with epiweeks on the x-azis and number of samples collected (n) on the y-axis</p>
<pre><code>geom_bar(aes(x=epiweek2, y=n),stat="identity", fill="darkgray",
         colour="darkgray")+
</code></pre>
<p>To create three line graphs on top on the bar graph which show the average time it takes for samples to be processed (meanpro),
samples to be quantified in the lab (meanlab) as well as the overall time (meanoverall). The mean turnaround times will be 
displayed on a secondary y-axis. Because the number of days is generally less than the number of samples collected, it affects 
the scale between the two y-axes. We can then multiply the values of the y-axis by 10 to ensure that they will be visible when plotting.</p>
<pre><code>geom_line(aes(x=epiweek2, y=Meanpro*10, colour= "Meanpro", group=1),stat="identity", size=1.5)+
geom_line(aes(x=epiweek2, y=Meanlab*10, colour= "Meanlab", group=1),stat="identity", size=1.5)+
geom_line(aes(x=epiweek2, y=Meanoverall*10, colour= "Meanoverall", group=1),stat="identity", size=1.5)+
</code></pre>
<p>To manually change the colours and labels of each line graph produced we can used the scale_colour_manual() function</p>
<pre><code>scale_colour_manual(labels = c("Sample Preparation", "Quantification", "Overall"), 
                  breaks = c("Meanpro", "Meanlab","Meanoverall" ),
                  values = c("forestgreen", "black", "skyblue")) +


scale_y_continuous(sec.axis=sec_axis(~ . /10,name="Mean TAT in days\n")) + 
labs(x="Epiweek",y="Number of samples collected\n")+
ggthemes::theme_hc()+
theme(
axis.ticks.x= element_blank(), 
axis.text.x = element_text(angle = 90, hjust = 0, size = 20),
axis.text.y = element_text(size = 20),
legend.position="bottom",
legend.title = element_blank(),
text = element_text( size=20),
axis.line.x = element_line(color="black", size = 1),
axis.line.y = element_line(color="black", size = 1)) +
geom_text(aes(label = n, x=epiweek2, y=n), vjust = -0.5, colour = "black", size = 6)

tat_plot

dev.off()
</code></pre>
<p><img alt="Screenshot" src="../img/quantitative.png" /></p>
<h1 id="calculating-turnaround-times-for-sequencing-results">Calculating turnaround times for sequencing results</h1>
<p>Since only PCR-positive samples are submitted for sequencing, we first filter out samples that were not sent for sequencing</p>
<pre><code>water &lt;- water %&gt;% 
filter(!is.na(water$date_sub_sequencing))
</code></pre>
<p>We then calculate TAT for each sample similar to what was done above </p>
<pre><code>seq_water$extraction_tat &lt;- difftime(seq_water$`Extraction Date`, 
                                 seq_water$`Date tested at Laboratory`
                                 , units = c("days"))

seq_water$pcr_tat &lt;- difftime(seq_water$`PCR Date`, 
                          seq_water$`Extraction Date`, 
                          units = c("days"))

seq_water$tapestation_tat &lt;- difftime(seq_water$`Tapestation Date`, 
                                  seq_water$`PCR Date`, 
                                  units = c("days"))

seq_water$sequencing_tat &lt;- difftime(seq_water$`Date received from Sequencing`,
                                 seq_water$`Date submitted for Sequencing`, 
                                 units = c("days"))

seq_water$seq_report_tat &lt;- difftime(seq_water$`Report Date`,
                                 seq_water$`Date received from Sequencing`, 
                                 units = c("days"))

seq_water$overall &lt;- difftime(seq_water$`Report Date` ,seq_water$`Date tested at Laboratory`, units = c("days"))
</code></pre>
<p>This is then used to calculate the mean TAT for each epiweek</p>
<pre><code>mean_extraction_tat &lt;- seq_water%&gt;% # we specify which df we would like to use
group_by(epiweek2)%&gt;% # we specify what we would like to group by 
summarise(Extraction=mean(extraction_tat, na.rm = TRUE))

mean_pcr_tat &lt;- seq_water%&gt;% # we specify which df we would like to use
group_by(epiweek2)%&gt;% # we specify what we would like to group by 
summarise(PCR=mean(pcr_tat, na.rm = TRUE))

mean_tapestation_tat &lt;- seq_water%&gt;% # we specify which df we would like to use
group_by(epiweek2)%&gt;% # we specify what we would like to group by 
summarise(Tapestation=mean(tapestation_tat, na.rm = TRUE))

mean_sequencing_tat &lt;- seq_water%&gt;% # we specify which df we would like to use
group_by(epiweek2)%&gt;% # we specify what we would like to group by 
summarise(Sequencing=mean(sequencing_tat, na.rm = TRUE))

mean_seq_report_tat &lt;- seq_water%&gt;% # we specify which df we would like to use
group_by(epiweek2)%&gt;% # we specify what we would like to group by 
summarise(`Sequencing Report`=mean(seq_report_tat, na.rm = TRUE))

mean_overall &lt;- seq_water%&gt;% # we specify which df we would like to use
group_by(epiweek2)%&gt;% # we specify what we would like to group by 
summarise(Overall=mean(overall, na.rm = TRUE))
</code></pre>
<p>We then tabulate the number of samples received by epiweek </p>
<pre><code>freq_samples2 &lt;- seq_water %&gt;%
group_by(epiweek2)%&gt;%
count(epiweek2, na.rm=TRUE)
</code></pre>
<p>A new dataframe is then created with the mean TAT and number of samples received. Since the merge function only takes two dataframes at time, we merge two dataframes at a time</p>
<pre><code>seqtat_vs_freq &lt;- merge(mean_extraction_tat, mean_pcr_tat, by="epiweek2", na.rm =TRUE)
seqtat_vs_freq &lt;- merge(seqtat_vs_freq, mean_tapestation_tat, by="epiweek2", na.rm =TRUE)
seqtat_vs_freq &lt;- merge(seqtat_vs_freq, mean_sequencing_tat, by="epiweek2", na.rm =TRUE)
seqtat_vs_freq &lt;- merge(seqtat_vs_freq, mean_seq_report_tat, by="epiweek2", na.rm =TRUE)
seqtat_vs_freq &lt;- merge(seqtat_vs_freq, mean_overall, by="epiweek2", na.rm =TRUE)
seqtat_vs_freq &lt;- merge(seqtat_vs_freq, freq_samples2, by="epiweek2", na.rm =TRUE)
</code></pre>
<p>A plot may now be created showing the TAT for each laboratory process</p>
<pre><code>png("/path/to/file/sequencing.png", 
    width = 5*900,
    height = 5*500, 
    res = 300,
    pointsize = 8)

tat_plot2 &lt;- ggplot(seqtat_vs_freq) +
geom_bar(aes(x=epiweek2, y=n),stat="identity", fill="darkgray",
       colour="darkgray")+
geom_line(aes(x=epiweek2, y=Extraction, colour= "Extraction", group=1),stat="identity", size=1.5) +
geom_line(aes(x=epiweek2, y=PCR, colour= "PCR", group=1),stat="identity", size=1.5) +
geom_line(aes(x=epiweek2, y=Tapestation, colour= "Tapestation", group=1),stat="identity", size=1.5) +
geom_line(aes(x=epiweek2, y=Sequencing, colour= "Sequencing", group=1),stat="identity", size=1.5)+
geom_line(aes(x=epiweek2, y=`Sequencing Report`, colour= "Sequencing Report", group=1),stat="identity", size=1.5)+
geom_line(aes(x=epiweek2, y=Overall, colour= "Overall", group=1),stat="identity", size=1.5)+
scale_y_continuous(sec.axis=sec_axis(~ . ,name="Mean TAT in days\n"))+ 
labs(x="Epiweek",y="Number of samples collected\n")+
ggthemes::theme_hc()+
theme(
    axis.ticks.x= element_blank(), 
    axis.text.x = element_text(angle = 90, hjust = 0, size = 20),
    axis.text.y = element_text(size = 20),
    legend.position="bottom",
    legend.title = element_blank(),
    text = element_text( size=20),
    axis.line.x = element_line(color="black", size = 1),
    axis.line.y = element_line(color="black", size = 1)) +
geom_text(aes(label = n, x=epiweek2, y=n), vjust = -0.5, colour = "black", size = 6)

tat_plot2

dev.off()
</code></pre>
<p><img alt="Screenshot" src="../img/sequencing.png" /></p>












                
              </article>
            </div>
          
          
  <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var tab,labels=set.querySelector(".tabbed-labels");for(tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script>

<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"/></svg>
  Back to top
</button>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      Copyright (c) 2024, NICD-Wastewater-Genomics
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    <script id="__config" type="application/json">{"base": "..", "features": ["navigation.tabs", "navigation.sections", "navigation.expand", "toc.integrate", "navigation.top", "search.suggest", "search.highlight", "content.tabs.link", "content.code.annotation", "content.code.copy"], "search": "../assets/javascripts/workers/search.b8dbb3d2.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../assets/javascripts/bundle.fe8b6f2b.min.js"></script>
      
    
  </body>
</html>